### PR TITLE
Let's upgrade to node 6

### LINKED ISSUE #56: Need updated hubot-slack once they release 4.0.2
The hubot slack adapter has had a fix for usernames with . or - merged to master, but not part part of a release yet.

https://github.com/slackhq/hubot-slack/pull/310

Thanks

### LINKED ISSUE #68: Use Node.js v6.x
Node.js v4.x goes into maintenance mode as of 2017-04-01. 

We should update our packaging to use Node.js v6.x by default.

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
closes https://github.com/StackStorm/st2chatops/issues/68
fixes https://github.com/StackStorm/st2chatops/issues/56

This should be merged only when we are ready to release st2 2.4.0 

* [X] hubot-slack 4.3.x https://github.com/StackStorm/hubot-stackstorm/pull/138
  - [x] Bump `hubot-stackstorm` version to `0.6`
* [X] Install docs update https://github.com/StackStorm/st2docs/pull/559
* [X] curl | bash installer changes https://github.com/StackStorm/st2-packages/pull/473
* [X] ansible-st2 installer changes https://github.com/StackStorm/ansible-st2/pull/154
* [X] Tested on U16 and CentOS7. Will rely on CI for U14 and EL6.

### CODE REVIEW

**Comment by arm4b** [comment]
Upgrading to node 6 will also pull the following: 
- Changing the node version in `st2chatops` package deb/rpm dependencies
- Changing node version in curl|bash installer
- Changing the node version in ansible-st2 installer
- st2docs update for manual installation steps
- https://github.com/StackStorm/st2-packages/issues/437 probably can be closed

first ideas, but could be also something else ¯\_(ツ)_/¯

File: debian/st2chatops.service [hunk 1]
```diff
@@ -14,6 +14,8 @@ TimeoutSec=60
 PrivateTmp=true
 Restart=on-failure
 RestartSec=5
+StandardOutput=syslog+console
+StandardError=syslog+console
```
Comment (arm4b) [inline debian/st2chatops.service:18]: Does that fix hubot logging bug described here: https://github.com/StackStorm/st2-packages/issues/300 ?

Comment (lakshmi-kannan) [reply debian/st2chatops.service:18]: It does not. This directive instructs journalctl to write to console which should get logged to /var/log according to docs but it doesn't. So sad.

Comment (arm4b) [reply debian/st2chatops.service:18]: With this param, feels like the truth is out there.

Let's try this for `st2chatops.service`: 
```diff
- ExecStart=/opt/stackstorm/chatops/bin/hubot $DAEMON_ARGS
+ ExecStart=exec bin/hubot $DAEMON_ARGS >> /var/log/st2/st2chatops.log 2>&1
```

File: package.json [hunk 2]
```diff
@@ -19,9 +19,9 @@
     "hubot-xmpp"        : "git+https://github.com/markstory/hubot-xmpp.git#94c3438e42778c53e38f6909939c514da50a0dca",
     "hubot-hipchat"     : "git+https://github.com/StackStorm/hubot-hipchat#sonnyp-patch",
     "hubot-irc"         : "git+https://github.com/nandub/hubot-irc#51d1f4b418fc039a456b7891c7f9bf836699cfdc",
-    "hubot-slack"       : "^3.4.2"
+    "hubot-slack"       : "^4.3.4"
```
Comment (arm4b) [inline package.json]: From the reports in GitHub https://github.com/StackStorm/st2chatops/issues/56#issuecomment-285562378, https://github.com/StackStorm/st2docs/issues/418 and community, this should also fix proxy support for Slack adapter :+1: 
cc @ytjohn

File: debian/control [hunk 3]
```diff
@@ -11,7 +11,7 @@ Vcs-Browser: https://github.com/stackstorm/st2chatops
 Package: st2chatops
 Architecture: any
 Pre-Depends: dpkg (>= 1.16.16)
-Depends: nodejs (>= 4.0.0)
+Depends: nodejs (>= 6.0.0)
```
Comment (arm4b) [inline debian/control]: Considering `nodejs` epopeya https://github.com/StackStorm/ops-infra/pull/206, please pin node versions to `6` for both deb/rpm.
So we won't install any node7 or node8 or whatever in future.

**Review by arm4b — CHANGES_REQUESTED** [review]

**Review by arm4b — APPROVED** [review]

**Pull request merged**